colossus/src/main/scala/colossus/core/Server.scala

```diff
@@ -64,6 +65,32 @@ case class ServerSettings(
   def highWatermark = highWatermarkPercentage * maxConnections
 }
 
+object ServerSettings {
+  def extract(config : Config) : ServerSettings = {
+    import colossus.metrics.ConfigHelpers._
+
+    //one-offing these, since they are going away soon
+    val bindingAttemptInterval = config.getFiniteDuration("binding-attempt-interval")
+    val bindingAttemptMaxTries = config.getLongOption("binding-attempt-max-tries")
+    val binding = PollingDuration(bindingAttemptInterval, bindingAttemptMaxTries)
+    val delegatorCreationInterval = config.getFiniteDuration("delegator-creation-interval")
+    val delegatorCreationMaxTries = config.getLongOption("delegator-creation-max-tries")
+    val delegator = PollingDuration(delegatorCreationInterval, delegatorCreationMaxTries)
+
```

Comment:
Without bringing in a third party lib, currying the constructor and just chaining applies seemed to be the cleanest and most readable way of doing this.